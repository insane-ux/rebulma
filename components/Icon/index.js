// @flow

import React from 'react'
import cn from 'classnames'

const Icon = ({ name, className }: { name: string, className?: string }) => (
  <span className={cn('icon', className)}>
    <i className={cn('fa', name)} />
  </span>
)

Icon.defaultProps = {
  className: undefined,
}

export default Icon
