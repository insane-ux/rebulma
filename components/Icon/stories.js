// @flow

import React from 'react'
import { storiesOf } from '@kadira/storybook'
import Icon from './'

storiesOf('Icon', module)
  .add('default', () => (
    <Icon name="fa-check" />
  ))
  .add('small', () => (
    <Icon className="is-small" name="fa-check" />
  ))
  .add('large', () => (
    <Icon className="is-large" name="fa-check" />
  ))
