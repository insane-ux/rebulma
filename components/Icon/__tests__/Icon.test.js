/* global it, expect, describe */

// @flow

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Icon from '../index'

describe('With Enzyme', () => {
  it('Icon with className', () => {
    const input = shallow(
      <Icon name="fa-check" className="is-small" />,
    )

    // The wrapper
    expect(input.find('.is-small').prop('className')).toEqual('icon is-small')
    // The Icon
    expect(input.find('i').prop('className')).toEqual('fa fa-check')
  })
})

describe('With Snapshot Testing', () => {
  it('Icon with className', () => {
    const component = renderer.create(<Icon name="fa-check" className="some" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
