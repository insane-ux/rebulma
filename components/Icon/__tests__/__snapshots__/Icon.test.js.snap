// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Icon with className 1`] = `
<span
  className="icon some"
>
  <i
    className="fa fa-check"
  />
</span>
`;
